app/src/main/java/com/bogdan/codeforceswatcher/features/problems/redux/requests/ProblemsRequests.kt

```diff
@@ -29,17 +34,28 @@ class ProblemsRequests {
             val problemsRu = promiseProblemsRu.await()?.body()?.result?.problems
 
             if (problemsEn == null || problemsRu == null) {
-                store.dispatch(Failure(null))
+                dispatchFailure()
             } else {
                 if (isProblemsMatching(problemsEn, problemsRu)) {
-                    store.dispatch(Success(mergeProblems(problemsEn, problemsRu)))
+                    val problems = mergeProblems(problemsEn, problemsRu)
```

Comment:
You can do all these 3 lines in `withContext(Dispatchers.IO)` and then dispatch Success after.